### PR TITLE
Created note listing page UI

### LINKED ISSUE #3: Create note listing page UI

### PR DESCRIPTION
**Pull request by gaagul**
Fixes #3

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Notes/Menu.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+
+// @ts-ignore
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Menu.jsx:3]: Be careful now. I had put up a lot of review comments in the Slack thread. The point of adding them in that thread is to ensure that I don't have to again provide redundant review comments. Go through the thread very well and make necessary changes where ever required.

File: app/javascript/src/components/Dashboard/Notes/NoteCard.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+import dayjs from "utils/dayjs";
+
+const NoteCard = ({ title, description, setShowDeleteAlert, createdAt }) => (
+  <div className="my-3 flex w-full flex-col border border-solid border-gray-300 p-4 shadow">
+    <div className="flex justify-between">
+      <Typography style="h4">{title}</Typography>
+      <Dropdown buttonStyle="text" icon={MenuVertical}>
+        <li>Edit</li>
+        <li onClick={setShowDeleteAlert}>Delete</li>
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/NoteCard.jsx:14]: Go through the Slack thread.

File: app/javascript/src/components/Dashboard/Notes/NoteCard.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      </Dropdown>
+    </div>
+    <div>
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/NoteCard.jsx:17]: Why do we need this div?

File: app/javascript/src/components/Dashboard/Notes/NoteCard.jsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      </Typography>
+    </div>
+    <hr className="my-2" />
+    <div className="flex justify-between">
+      <Tag label="Getting Started" />
+      <div className="flex items-center">
+        <Clock size={14} />
+        <Tooltip content={dayjs().format("dddd, hh:mmA")} position="bottom">
+          <Typography className="px-1" style="body3">
+            {`Created ${dayjs(createdAt).fromNow()}`}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/NoteCard.jsx:29]: Move these into utils.

File: app/javascript/src/components/Dashboard/Notes/NoteCard.jsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+          </Typography>
+        </Tooltip>
+        <Avatar
+          user={{
+            name: "neeto UI",
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/NoteCard.jsx:34]:
```suggestion
            name: "neetoUI",
```

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 6]
```diff
@@ -43,65 +43,59 @@ const Notes = () => {
...
-          image={EmptyNotesListImage}
-          primaryAction={() => setShowNewNotePane(true)}
-          primaryActionLabel="Add new note"
-          subtitle="Add your notes to send customized emails to them."
-          title="Looks like you don't have any notes!"
+    <>
+      <Menu showMenu={showMenuBar} />
+      <Container>
+        <Header
+          menuBarToggle={() => setShowMenuBar(!showMenuBar)}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:50]: https://academy.bigbinary.com/learn-react/pages/know-when-to-use-previous-state-for-state-update

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 7]
```diff
@@ -43,65 +43,59 @@ const Notes = () => {
...
-        <DeleteAlert
-          refetch={fetchNotes}
-          selectedNoteIds={selectedNoteIds}
-          setSelectedNoteIds={setSelectedNoteIds}
-          onClose={() => setShowDeleteAlert(false)}
+        {notes.length ? (
+          notes.map(note => (
+            <NoteCard
+              createdAt={note["createdAt"]}
+              description={note["description"]}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:69]: Use `dot` notation.

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 8]
```diff
@@ -43,65 +43,59 @@ const Notes = () => {
...
+          setShowPane={setShowNewNotePane}
+          showPane={showNewNotePane}
         />
-      )}
-    </Container>
+        {showDeleteAlert && (
+          <DeleteAlert
+            refetch={fetchNotes}
+            selectedNoteIds={[]}
+            setSelectedNoteIds={[]}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:93]: Why is this passed with an empty array when the value is supposed to be a function?

File: app/javascript/src/utils/dayjs.js [hunk 9]
```diff
@@ -0,0 +1,6 @@
+import dayjs from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
+
+dayjs.extend(relativeTime);
+
+export default dayjs;
```
Comment (yedhink) [inline app/javascript/src/utils/dayjs.js:6]: https://academy.bigbinary.com/learn-react/pages/miscellaneous-tips#:~:text=How%20to%20extend%20day.js%3F

**Review by yedhink — CHANGES_REQUESTED** [review]
@gaagul

File: app/javascript/src/components/Dashboard/Notes/index.jsx (hunk 8 above)
Comment (gaagul) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:93]: Sorry for leaving it like that, I was hoping to deal with the whole delete section while Solving issue 4.
I will make sure that these mistakes won't happen from now on.

File: app/javascript/src/components/Dashboard/Notes/NoteCard.jsx (hunk 2 above)
Comment (gaagul) [reply app/javascript/src/components/Dashboard/Notes/NoteCard.jsx:14]: ` const handleDelete = () => {
    setShowDeleteAlert(true);
    setSelectedNote(note);
  };`
.
.
.
.
` <li onClick={handleDelete}>Delete</li> `

I have created a handledelete function inside NotesCard component and passed it in the onClick attribute,
Is this the correct way to solve the above-mentioned problem in the code?

Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/NoteCard.jsx:14]: Don't think you've gone very well through the Slack thread. https://bigbinary.slack.com/archives/C01T5Q4DZAR/p1663761829787419?thread_ts=1663060378.971029&cid=C01T5Q4DZAR

Comment (gaagul) [reply app/javascript/src/components/Dashboard/Notes/NoteCard.jsx:14]: Sorry for that silly mistake from my side,
I have changed the list to MenuItem.Button and also went through the Semantic HTML and accessibility part in the handbook once again to make sure that I understood it perfectly.

**Pull request merged**